### DISCUSSION TITLE
How to tag faces with Digikam and read them in Pigallery2?

### DISCUSSION THREAD

**Discussion by v-marinkov** [bpatrik/pigallery2#644] [Q&A] [answered]
I tag faces in Digikam using the face regions and I have enabled writing to XMP sidecards. In Pigallery2, however, names only come up as tags and searching for `people:"First Last"` yields no results. What is the correct way to tag faces, so that Pigallery2 can read them?

**Comment by bpatrik** [comment, marked as answer]
Sidecars are not yet supported. You need to write the xmp data to the image
file itself.

-- Sorry for being brief, sent from my phone.

**Reply by v-marinkov** [reply]
Thanks for the quick reply, but even then - how should faces be written so Pigallery can recognise them? If I use the face rectangle/detection function of Digikam, Pigallery just sees tags for the name, not even "People/Name". What fields is "people: keyword" searching for and how to populate the Faces tab of Pigallery.

**Reply by v-marinkov** [reply]
Never mind, I had to ask Digikam to save faces data to files by checking the option  Configure / Metadata / Behaviour / Face Tags (including face areas)

**Comment by hajo62** [comment]
I did the same for many of my photos.
Suddenly the Faces-_tab_ does not display faces anymore, but the single photos still have and show the faces tags.
Maybe you have an idea?

I have opened an issue #656 for this...

**Comment by nk9** [comment]
There's a setting called "Face Tags (including face areas)" in the Metadata section of digiKam. (See [DK docs](https://docs.digikam.org/en/setup_application/metadata_settings.html).) If you enable this, DK will start to write the face regions to the files themselves. That way, they will be available for piGallery2.

One caveat: DK doesn't seem to save these regions to the files unless you _create them with the setting on._ Sometimes the rectangles are saved when you use "Album > Write Metadata to Files", but it seems very hit-or-miss.

**Comment by rhatguy** [comment]
Adding this here in case anyone else comes looking for a similar solution.  You can use exiftool to copy faces from XMP files into the original JPG with a command like the below.  I previously used digikam but have since moved to Mylio for the actual face tagging.  Mylio only writes to XMP, but I can simply copy any face written by Mylio back into the JPG and PiGallery2 reads them without any issues.

exiftool.exe -r -ext jpg -overwrite_original -tagsfromfile %d%f.xmp -all:all -wm cg c:\test
